packages/api-graphql/src/types/index.ts

```diff
@@ -14,10 +14,12 @@
 import { GraphQLError } from 'graphql/error/GraphQLError';
 import { DocumentNode } from 'graphql/language/ast';
 
+type GRAPHQL_STR_MODE = 'API_KEY' | 'AWS_IAM' | 'OPENID_CONNECT' | 'AMAZON_COGNITO_USER_POOLS' | 'AWS_LAMBDA';
+
 export interface GraphQLOptions {
 	query: string | DocumentNode;
 	variables?: object;
-	authMode?: GRAPHQL_AUTH_MODE;
+	authMode?: GRAPHQL_AUTH_MODE | GRAPHQL_STR_MODE;
```

Comment:
No need to create a new type. You can accomplish the same thing with `keyof typeof` for example:
```suggestion
	authMode?: GRAPHQL_AUTH_MODE | keyof typeof GRAPHQL_AUTH_MODE;
```

Reply:
Awesome, thank you Ivan! :)

Reply:
Thank you for the feedback! I will make the change.